fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/load/ChunkWriter.java

```diff
@@ -248,4 +261,68 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
+     * 1 - the shared cache would have to be updated at the end of a transaction (we don't control it).
+     * 2 - we would have to use a transaction sync registry to control the synchronization of the cache.
+     * 3 - Instead, we're doing a read then update.
+     *
+     * @param chunkData the transient user data used increment the number of skips
+     * @param skip should skip the resource if it matches
+     * @param localCache map containing the key-saltHash
+     * @param persistence used to facilitate the calls to the underlying db
+     * @param context used in db calls
+     * @param logicalId the logical id of the FHIR resource (e.g. 1-2-3-4)
+     * @param resource the FHIR Resource
+     * @return outcomes including information or warnings
+     * @throws FHIRPersistenceException
+     */
+    public OperationOutcome conditionalFingerprintUpdate(ImportTransientUserData chunkData, boolean skip, Map<String, SaltHash> localCache, FHIRPersistence persistence, FHIRPersistenceContext context, String logicalId, Resource resource) throws FHIRPersistenceException {
+        // Key is scoped to the ResourceType.
+        String key = resourceType + "/" + logicalId;
+        SaltHash oldBaseLine = localCache.get(key);
+
+        ResourceFingerprintVisitor fp = new ResourceFingerprintVisitor();
+        if (oldBaseLine == null) {
+            // Go get the latest resource in the database and fingerprint the resource.
+            // If the resource exists, then we need to fingerprint.
+            Resource oldResource = persistence.read(context, resource.getClass(), logicalId).getResource();
+            if (oldResource != null) {
+                ResourceFingerprintVisitor fpOld = new ResourceFingerprintVisitor();
+                oldResource.accept(fpOld);
+                oldBaseLine = fpOld.getSaltAndHash();
+                fp = new ResourceFingerprintVisitor(oldBaseLine);
+            }
+        }
+
+        resource.accept(fp);
+        SaltHash newBaseLine = fp.getSaltAndHash();
+
+        OperationOutcome oo;
+        if (oldBaseLine != null && oldBaseLine.equals(newBaseLine) && skip) {
+            if (logger.isLoggable(Level.FINE)) {
+                logger.fine("Skipping $import - update for '" + key + "'");
+            }
+            chunkData.addToNumOfSkippedResources(1);
+            oo =  OperationOutcome.builder()
+                .issue(Issue.builder()
+                    .severity(IssueSeverity.INFORMATION)
+                    .code(IssueType.INFORMATIONAL)
+                    .details(CodeableConcept.builder()
+                        .text(string("Update resource matches the existing resource; skipping the update for '" + key + "'"))
+                        .build())
+                    .build())
+                .build();
+        } else {
+            // We need to update the db and update the local cache
+            oo = persistence.update(context, logicalId, resource).getOutcome();
```

Comment:
I'm wondering if we should add the previously-read "oldResource" to the PersistenceEvent within context.
If the event is missing this prevResource then the JDBC impl will do yet another read to get it:
```
            // If the "previous resource" is set in the persistence event, then get the
            // existing version # from that.
            if (context.getPersistenceEvent() != null && context.getPersistenceEvent().isPrevFhirResourceSet()) {
                Resource existingResource = context.getPersistenceEvent().getPrevFhirResource();
                if (existingResource != null) {
                    log.fine("Using pre-fetched 'previous' resource.");
                    String version = existingResource.getMeta().getVersionId().getValue();
                    existingVersion = Integer.valueOf(version);
                }
            }

            // Otherwise, go ahead and read the resource from the datastore and get the
            // existing version # from it.
            else {
                log.fine("Fetching 'previous' resource for update.");
                existingResourceDTO = resourceDao.read(logicalId, resourceType.getSimpleName());
                if (existingResourceDTO != null) {
                    existingVersion = existingResourceDTO.getVersionId();
                }
            }
```

Reply:
I've added with an if-else

Reply:
looks good